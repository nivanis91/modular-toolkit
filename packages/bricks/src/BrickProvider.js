import React, { Component } from 'react';
import BrickManagerContext from './BrickManagerContext';
import BrickManager from './BrickManager';
import PropTypes from 'prop-types';

class BrickProvider extends Component {
    constructor(props) {
        super(props);
        const { store, reducer, sagaMiddleware } = this.props;
        this.state = {
            brickManager: new BrickManager({ store, reducer, sagaMiddleware })
        };
    }

    render() {
        const { children } = this.props;
        const { brickManager } = this.state;
        return <BrickManagerContext.Provider value={brickManager}>{children}</BrickManagerContext.Provider>;
    }
}

BrickProvider.propTypes = {
    store: PropTypes.object.isRequired,
    reducer: PropTypes.func,
    sagaMiddleware: PropTypes.func,
    children: PropTypes.node
};

export default BrickProvider;
