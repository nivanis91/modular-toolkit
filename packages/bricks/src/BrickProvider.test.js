import React from 'react';
import BrickProvider from './BrickProvider';
import { mount } from 'enzyme';
import BrickManagerContext from './BrickManagerContext';

const sagaMiddleware = jest.fn();
const store = {};
const reducer = jest.fn();
const mockBrickManagerConstructor = jest.fn();
const consumer = jest.fn();
jest.mock(
    './BrickManager',
    () =>
        class BrickManager {
            constructor(...args) {
                mockBrickManagerConstructor(...args);
            }
        }
);

describe('When I mount a brick provider component', () => {
    beforeEach(() =>
        mount(
            <BrickProvider sagaMiddleware={sagaMiddleware} store={store} reducer={reducer}>
                <BrickManagerContext.Consumer>{consumer}</BrickManagerContext.Consumer>
            </BrickProvider>
        ));
    describe('a Brick Manager', () =>
        void it('is instantiated with the provided saga middleware, store and reducer', () =>
            expect(mockBrickManagerConstructor).toBeCalledWith({
                store,
                reducer,
                sagaMiddleware
            })));
    describe('the consumer wrapped by the brick provider component', () =>
        void it('has access to the brick manager', () => expect(consumer.mock.calls[0]).toMatchSnapshot()));
    afterEach(() => jest.clearAllMocks());
});
