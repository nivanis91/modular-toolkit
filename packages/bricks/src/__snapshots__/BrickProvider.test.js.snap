// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I mount a brick provider component the consumer wrapped by the brick provider component has access to the brick manager 1`] = `
Array [
  BrickManager {},
]
`;
